vendor/plugins/edu.harvard.dce.paella.singleVideoToggle/singleVideoToggle.js

```diff
@@ -10,40 +11,76 @@ paella.addPlugin(function () {
       this._masterVideo = null;
       this._toggleIndex = 1; //toggle to presentation when button pressed first time
     }
     getDefaultToolTip () {
-      return base.dictionary.translate("Switch videos");
+      return base.dictionary.translate(this._toolTipOpts[this._getOptIndex()]);
     }
     getAriaLabel() {
-      return base.dictionary.translate("Switch videos");
+      return base.dictionary.translate(this._toolTipOpts[this._getOptIndex()]);
     }
     getAlignment() {
       return 'right';
     }
     getSubclass() {
-      return "showViewModeButton";
+      // #DCE OPC-497 show current res text on button (get the mutli quality button style)
+      return base.dictionary.translate(this._subClassOpts[this._getOptIndex()]);
     }
     getIconClass() {
-      return 'icon-presentation-mode';
+      // #DCE OPC-497 show current res text on button
+      return base.dictionary.translate(this._iconClassOpts[this._getOptIndex()]);
     }
     getIndex() {
       return 450;
     }
     getInstanceName() {
       return "singleVideoTogglePlugin";
     }
-    getName () {
+    getName() {
       return "edu.harvard.dce.paella.singleVideoTogglePlugin";
     }
-    _currentPlayerProfile () {
+
+    _getOptIndex() {
+        return (paella.dce && paella.dce.hlsLiveToggleV1) ? this._toggleHlsLiveResV1Index : this._toggleSingleVideoIndex;
+    }
+
+    _currentPlayerProfile() {
       return paella.player.selectedProfile;
     }
+
+    _printResIfHlsLive(streams) {
+      // Check if hlsLiveToggleV1 flag exists and the path to the source resolution value exists
+      // Then add the current resolution value into the control button
+      if  (paella.dce && paella.dce.hlsLiveToggleV1 && streams
+          && streams.length > 0
+          && streams[0].sources
+          && streams[0].sources.hls
+          && streams[0].sources.hls.length > 0
+          && streams[0].sources.hls[0].res) {
+            this.setText(streams[0].sources.hls[0].res.w + "p"); // change the UI res text
+      }
+    }
+
     checkEnabled (onSuccess) {
-      // Only enable for iOS (not Android) TODO: test with Safari on Android?
-      onSuccess (base.userAgent.system.iOS && paella.dce && paella.dce.sources && paella.dce.sources.length > 1 && ! paella.dce.blankAudio);
+      // Enable for iOS (not Android) TODO: test with Safari on Android?
+      var enabled = base.userAgent.system.iOS && paella.dce && paella.dce.sources && paella.dce.sources.length > 1 && ! paella.dce.blankAudio;
+
+      // #DCE OPC-497 also enable for HLS Live when 2 seperate m3u8 passed with different res for the same video, instead of one master m3u8 containing 2 res.
+      if (paella.dce && paella.dce.hlsLiveToggleV1) {
+          this._printResIfHlsLive(paella.player.videoLoader._data.streams);
+          this._curOptIndex = this._toggleHlsLiveResV1Index; // tool tip and class uses HLS live text and classes
+          enabled = true;
+      } // end DCE HLS Live v1
```

Comment:
Hey, there is a "var" (not "let"), otherwise looks good.

Reply:
The gulp build libs are older and don't like var!  I would rather use let, but it's upstream gulp build

Reply:
Oh, actually that might have been a different area. I'll change to let.

Reply:
fixed!